Compiling current master fails

Result of mvn clean install using JDK 1.8:
[INFO] --- spotbugs-maven-plugin:3.1.12.2:check (spotbugs) @ test-notifications ---
[INFO] BugInstance size is 1
[INFO] Error size is 0
[INFO] Total bugs: 1
[ERROR] Possible null pointer dereference in de.tum.in.www1.jenkins.notifications.SendTestResultsNotificationPostBuildTask.perform(Run, FilePath, Launcher, TaskListener) due to return value of called method [de.tum.in.www1.jenkins.notifications.SendTestResultsNotificationPostBuildTask, de.tum.in.www1.jenkins.notifications.SendTestResultsNotificationPostBuildTask] Method invoked at SendTestResultsNotificationPostBuildTask.java:[line 78]Known null at SendTestResultsNotificationPostBuildTask.java:[line 79] NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE

--- a/pom.xml
+++ b/pom.xml
@@ -9,7 +9,7 @@
     </parent>
     <groupId>io.jenkins.plugins</groupId>
     <artifactId>test-notifications</artifactId>
-    <version>1.2.0</version>
+    <version>1.2.1</version>
     <packaging>hpi</packaging>
     <properties>
         <!-- Baseline Jenkins version you use to build the plugin. Users must have this version or newer to run. -->
@@ -85,7 +85,7 @@
         <dependency>
             <groupId>de.tum.in.ase</groupId>
             <artifactId>static-code-analysis-parser</artifactId>
-            <version>1.2.0</version>
+            <version>1.3.0</version>
         </dependency>
     </dependencies>
 

--- a/src/main/java/de/tum/in/www1/jenkins/notifications/HttpHelper.java
+++ b/src/main/java/de/tum/in/www1/jenkins/notifications/HttpHelper.java
@@ -13,10 +13,10 @@ import java.io.IOException;
 
 public class HttpHelper {
 
-    public static void postTestResults(TestResults results, String url, Secret token) throws IOException, HttpException {
+    public static void postTestResults(TestResults results, String url, String secret) throws IOException, HttpException {
         final String body = new Gson().toJson(results);
         final HttpResponse response = Request.Post(url)
-                .addHeader("Authorization", token.getPlainText())
+                .addHeader("Authorization", secret)
                 .addHeader("Accept", ContentType.APPLICATION_JSON.getMimeType())
                 .bodyString(body, ContentType.APPLICATION_JSON)
                 .execute()

--- a/src/main/java/de/tum/in/www1/jenkins/notifications/HttpHelper.java
+++ b/src/main/java/de/tum/in/www1/jenkins/notifications/HttpHelper.java
@@ -1,15 +1,16 @@
 package de.tum.in.www1.jenkins.notifications;
 
-import com.google.gson.Gson;
-import hudson.util.Secret;
-import de.tum.in.www1.jenkins.notifications.model.TestResults;
+import java.io.IOException;
+
 import org.apache.commons.io.IOUtils;
 import org.apache.http.HttpException;
 import org.apache.http.HttpResponse;
 import org.apache.http.client.fluent.Request;
 import org.apache.http.entity.ContentType;
 
-import java.io.IOException;
+import com.google.gson.Gson;
+
+import de.tum.in.www1.jenkins.notifications.model.TestResults;
 
 public class HttpHelper {
 

--- a/src/main/java/de/tum/in/www1/jenkins/notifications/SendTestResultsNotificationPostBuildTask.java
+++ b/src/main/java/de/tum/in/www1/jenkins/notifications/SendTestResultsNotificationPostBuildTask.java
@@ -12,7 +12,6 @@ import javax.xml.bind.JAXBContext;
 import javax.xml.bind.JAXBException;
 import javax.xml.bind.Unmarshaller;
 
-import de.tum.in.www1.jenkins.notifications.model.*;
 import jenkins.model.Jenkins;
 import jenkins.tasks.SimpleBuildStep;
 
@@ -32,6 +31,7 @@ import de.tum.in.ase.parser.ReportParser;
 import de.tum.in.ase.parser.domain.Report;
 import de.tum.in.ase.parser.exception.ParserException;
 import de.tum.in.www1.jenkins.notifications.exception.TestParsingException;
+import de.tum.in.www1.jenkins.notifications.model.*;
 
 import hudson.Extension;
 import hudson.FilePath;
@@ -47,7 +47,6 @@ import hudson.tasks.Publisher;
 import hudson.tasks.Recorder;
 import hudson.util.FormValidation;
 import hudson.util.ListBoxModel;
-import hudson.util.Secret;
 
 public class SendTestResultsNotificationPostBuildTask extends Recorder implements SimpleBuildStep {
 

--- a/src/main/java/de/tum/in/www1/jenkins/notifications/SendTestResultsNotificationPostBuildTask.java
+++ b/src/main/java/de/tum/in/www1/jenkins/notifications/SendTestResultsNotificationPostBuildTask.java
@@ -73,9 +73,9 @@ public class SendTestResultsNotificationPostBuildTask extends Recorder implement
         final List<Testsuite> testReports = extractTestResults(taskListener, testResultsDir);
         final List<Report> staticCodeAnalysisReport = parseStaticCodeAnalysisReports(taskListener, staticCodeAnalysisResultsDir);
         final TestResults results = combineTestResults(run, testReports, staticCodeAnalysisReport);
-        final Secret secret = Objects.requireNonNull(CredentialsProvider
-                .findCredentialById(credentialsId, StringCredentials.class, run, Collections.emptyList()))
-                .getSecret();
+        final StringCredentials credentials = CredentialsProvider
+                .findCredentialById(credentialsId, StringCredentials.class, run, Collections.emptyList());
+        final String secret = credentials != null ? credentials.getSecret().getPlainText() : "Credentials containing the Notification Plugin Secret not found";
 
         // Post results to notification URL
         try {

